services/apps/premium/members_enrichment_worker/src/activities/lf-auth0/authenticateLFAuth0.ts

```diff
@@ -0,0 +1,56 @@
+import { AuthenticationClient } from 'auth0'
+import { svc } from '../../main'
+import { RedisCache } from '@crowd/redis'
+import { ITokenWithExpiration } from '../../types/lfid-enrichment'
+
+export async function refreshToken(token?: string): Promise<string> {
+  const redisCache = new RedisCache(`lfx-auth0`, svc.redis, svc.log)
+  const tokenFromRedis = await redisCache.get('access-token')
+
+  if (!token) {
+    if (tokenFromRedis) {
+      return tokenFromRedis
+    }
+    const tokenFromAuth0 = await getTokenFromAuth0()
+    await saveTokenToRedis(tokenFromAuth0, redisCache)
+    return tokenFromAuth0.token
+  }
+
+  // if token is sent along with the function, check if sent token is same as redis token
+  if (token === tokenFromRedis) {
+    const tokenFromAuth0 = await getTokenFromAuth0()
+    await saveTokenToRedis(tokenFromAuth0, redisCache)
+    return tokenFromAuth0.token
+  }
```

Comment:
It's for multiple workers working on the same API - 
We send the current token when refreshing token - If token in redis and current token that we're working on are not same it means that it was already refreshed by some other worker, so we can use that instead of refreshing the token again

Reply:
1. I didn't see this function `refreshToken()` called with any arguments
2. > If token in redis and current token that we're working on are not same it means that it was already refreshed by some other worker, so we can use that instead of refreshing the token again

    Yes, this makes sense. This is the very last return statement in this function.

    But I'm talking about the case when the `token === tokenFromRedis` comparison is true, and we still fetch a new token from auth0. Shouldn't we just return passed `token` instead, because this is exactly what is already stored in redis?

Reply:
ur right - simplified the function a bit